packages/@aws-cdk/core/lib/private/cloudformation-lang.ts

```diff
@@ -204,3 +323,59 @@ export function isNameOfCloudFormationIntrinsic(name: string): boolean {
   // these are 'fake' intrinsics, only usable inside the parameter overrides of a CFN CodePipeline Action
   return name !== 'Fn::GetArtifactAtt' && name !== 'Fn::GetParam';
 }
+
+/**
+ * Separated iterator
+ */
+function* sepIter<A>(xs: Iterable<A>): IterableIterator<[boolean, A]> {
+  let comma = false;
+  for (const item of xs) {
+    yield [comma, item];
+    comma = true;
+  }
+}
+
+/**
+ * Object.entries() but skipping undefined values
+ */
+function* definedEntries<A extends object>(xs: A): IterableIterator<[string, any]> {
+  for (const [key, value] of Object.entries(xs)) {
+    if (value !== undefined) {
+      yield [key, value];
+    }
+  }
+}
+
+/**
+ * Quote string literals inside an intrinsic
+ */
+function quoteInsideIntrinsic(x: any): any {
```

Comment:
This feels awfully fragile. Any way to avoid having to encode a heuristic for each fn?

Reply:
Not really. We can encode it in a different place and make it more official, but fundamentally this is what we have to do.